MilitaryManager.Core/Services/DivisionService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Ardalis.Specification;
+using AutoMapper;
+using MilitaryManager.Core.DTO.Divisions;
+using MilitaryManager.Core.Entities.DivisionEntity;
+using MilitaryManager.Core.Entities.DivivsionEntity;
+using MilitaryManager.Core.Interfaces.Repositories;
+using MilitaryManager.Core.Interfaces.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace MilitaryManager.Core.Services
+{
+    public class DivisionService: IDivisionService
+    {
+        protected readonly IRepository<Division, int> _divisionRepository;
+        protected readonly IMapper _mapper;
+        public DivisionService(IRepository<Division, int> divisionRepository, IMapper mapper)
+        {
+            _divisionRepository = divisionRepository;
+            _mapper = mapper;
+        }
+        public async Task<IEnumerable<DivisionDTO>> GetAllDivisionsAsync()
+        {
+            Specification<Division> specification = new Divisions.DivisionsList();
+            var divisions = await _divisionRepository.GetListBySpecAsync(specification);
+            var getDivisions = divisions.Where(x => x.ParentId == null);
+
+            return _mapper.Map<IEnumerable<DivisionDTO>>(getDivisions);
+        }
+
+        public async Task<DivisionDTO> GetDivisionByKeyAsync(int id)
+        {
+            Specification<Division> specification = new Divisions.DivisionsList();
+            var division = await _divisionRepository.GetListBySpecAsync(specification);
```

Comment:
Use GetFirstBySpecAsync(specification)

Reply:
If I do it in such a way, I won't be able to see the subdivisions of subdivision